src/test/java/edu/ucsb/cs156/organic/controllers/CoursesControllerTests.java

```diff
@@ -141,6 +143,108 @@ public void user_can_get_only_courses_for_which_they_are_staff() throws Exceptio
+                 assertEquals("EntityNotFoundException", json.get("type"));
+                 assertEquals("Course with id 7 not found", json.get("message"));
+         }
+
+         @WithMockUser(roles = { "USER" })
+         @Test
+         public void test_that_instructor_can_get_by_id_when_the_id_exists() throws Exception {
+ 
+                User currentUser = currentUserService.getCurrentUser().getUser();
+
+                Staff courseStaff1 = Staff.builder()
+                                .id(1L)
+                                .courseId(course1.getId())
+                                .githubId(currentUser.getGithubId())
+                                .user(currentUser)
+                                .build();
+
+                when(courseRepository.findById(eq(course1.getId()))).thenReturn(Optional.of(course1));
+                when(courseStaffRepository.findByCourseIdAndGithubId(eq(course1.getId()),
+                                eq(courseStaff1.getGithubId()))).thenReturn(Optional.of(courseStaff1));
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/courses/get?id=1"))
+                                .andExpect(status().isOk()).andReturn();
+
+
+                // assert
+                verify(courseRepository, times(1)).findById(eq(1L));
+                verify(courseStaffRepository, times(1)).findByCourseIdAndGithubId(eq(course1.getId()),
+                                eq(courseStaff1.getGithubId()));
+                String expectedJson = mapper.writeValueAsString(course1);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+         }
+
+         @WithMockUser(roles = { "USER" })
+         @Test
+         public void test_that_non_admin_non_instructor_cannot_get_an_existing_course() throws Exception {
+ 
+                 // arrange
+                User currentUser = currentUserService.getCurrentUser().getUser();
+
+                when(courseRepository.findById(eq(course1.getId()))).thenReturn(Optional.of(course1));
+
+                // act
+                // MvcResult response = mockMvc.perform(get("/api/courses/get?id=1"))
+                //                 .andExpect(status().isForbidden()).andReturn();
+                mockMvc.perform(get("/api/courses/get?id=1"))
+                   .andExpect(status().isForbidden());
+
+
+                // assert
+                verify(courseRepository, times(1)).findById(eq(1L));
+                // Map<String, Object> json = responseToJson(response);
+                // System.out.print("json = ");
+                // System.out.println(json.toString());
+                // assertEquals("AccessDeniedException", json.get("type"));
+                // assertEquals(String.format("User %s is not authorized to get course 1", currentUser.getGithubLogin()), json.get("message"));
```

Comment:
It will speed things up a lot if you take care of this at an earlier stage in the process
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html